### PR TITLE
Add build pools, change gpu targets, and build tests

### PR DESCRIPTION
**Pull request by amd-jmacaran**
To best determine hardware specs per repo, added more build pool options with varying number of vCPUs, RAM size, etc. and will kick off builds with test targets enabled to determine long-term cost values.

pipelines_repo reference not necessary for tag-builds. Removing the references makes testing/debugging easier.

### CODE REVIEW

**Review by alexxu-amd — APPROVED** [review]

**Comment by amd-jmacaran** [comment]
After refactors, all components building with test binaries.

File: .azuredevops/tag-builds/AMDMIGraphX.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 variables:
 - group: common
-- template: /.azuredevops/variables-global.yml@pipelines_repo
```
Comment (alexxu-amd) [inline .azuredevops/tag-builds/AMDMIGraphX.yml:3]: We removed all pipelines_repo usage but still define it. Is it refactored somewhere?

File: .azuredevops/components/rocminfo.yml [hunk 2]
```diff
@@ -5,14 +5,18 @@ parameters:
...
+    - ROCR-Runtime
 
 jobs:
 - job: rocminfo
   variables:
   - group: common
   - template: /.azuredevops/variables-global.yml
   pool:
-    vmImage: ${{ variables.LOW_END_BUILD_POOL }}
+    vmImage: ${{ variables.BASE_BUILD_POOL }}
```
Comment (alexxu-amd) [inline .azuredevops/components/rocminfo.yml:19]: I think BASE_BUILD_POOL is a local machine. Wonder if that's desired

Comment (amd-jmacaran) [reply .azuredevops/components/rocminfo.yml:19]: BASE_BUILD_POOL is not a local machine. It is the ubuntu-22.04, Microsoft-hosted pool. This is desired. ON_PREM_BUILD_POOL are local machines.

File: .azuredevops/tag-builds/AMDMIGraphX.yml (hunk 1 above)
Comment (amd-jmacaran) [reply .azuredevops/tag-builds/AMDMIGraphX.yml:3]: Where do you see it defined? It is implied that the current repo is the pipelines_repo (i.e., ROCm/ROCm).

Comment (alexxu-amd) [reply .azuredevops/tag-builds/AMDMIGraphX.yml:3]: I'm a little confused about this, pipelines_repo seems like a variable defined by us. The first time it appears is in tag-builds/*.yml under resources section (please correct me if I'm wrong). So I thought this was the definition and later when we do @pipelines_repo, we specifying repo as ROCm/ROCm.

**Review by alexxu-amd — APPROVED** [review]

**Pull request merged**